test/test_issue655.py

```diff
@@ -0,0 +1,27 @@
+from rdflib import Graph, Namespace, URIRef, Literal
+from rdflib.compare import to_isomorphic
+import unittest
+
+
+class TestIssue655(unittest.TestCase):
```

Comment:
Sure, tests have been added in 3e9a908 (ignoring datatypes). And the solution was extended to NaN in e2c89dc.